frontend/public/components/utils/dropdown.jsx

```diff
@@ -35,9 +35,9 @@ export class DropdownMixin extends React.PureComponent {
     this.hide(event);
   }
 
-  componentWillReceiveProps({ selectedKey }) {
+  componentWillReceiveProps({ selectedKey, items }) {
     if (selectedKey !== this.props.selectedKey) {
-      this.setState({ selectedKey });
+      this.setState({ selectedKey, title: items[selectedKey] });
```

Comment:
I don't think it will be an issue since logic to handle new title prop is there on line 268, under `componentWillRecieveProps` of `Dropdown` component.

And `props.noSelection` is only relevant when user clicks on the items in dropdown